tests/testthat/test-utils-parse.R

```diff
@@ -327,12 +327,39 @@ test_that('parseWaterQualityMeasurements doesnt error when given invalid JSON',
                             "timeZone": "CST"
                             }
                             ]
-}')
+  }')
 
   reportObject2 <- fromJSON('{ }')
   
   expect_equal(repgen:::parseWaterQualityMeasurements(reportObject1), NULL)
   expect_equal(repgen:::parseWaterQualityMeasurements(reportObject2), NULL)
-  })
+})
+
+test_that('fetchExcludedControlConditions properly retrieves the excluded control condition data', {
+  controlConditionJSON <- fromJSON('{
+     "excludedControlConditions": [
+        {
+          "value": "Clear",
+          "name": "CLEAR"
+        },
+        {
+          "value": "VegetationLight",
+          "name": "VEGETATION_LIGHT"
+        },
+        {
+          "value": "VegetationModerate",
+          "name": "VEGETATION_MODERATE"
+        }
+     ]
+  }')
+  
+  conditions <- repgen:::parseExcludedControlConditions(controlConditionJSON)
+  
+  expect_warning(expect_equal(repgen:::parseExcludedControlConditions(c()), NULL))
```

Comment:
The warning given here is caused by the validate data function warning that there was no valid data returned. I suppose that's really testing the validateFetchedData function rather than this parse function so it's not really necessary here. I'll update that